scripts/tybalt_vae.py

```diff
@@ -0,0 +1,320 @@
+# applied to gene expression data across 33 different cancer-types from The Cancer Genome Atlas (TCGA).
+# 
+# A VAE aproximates the data generating function for the cancer data and learns the lower dimensional manifold a tumor occupies in gene expression space. By compressing the gene expression space into lower dimensional space, the VAE would, ideally, learn biological principles, such as cancer hallmark pathway activations, that help explain how tumors are similar and different. The VAE is also a generative model with a latent space that can be interpolated to observe transitions between cancer states.
+# 
+# The particular model trained in this notebook consists of gene expression input (5000 most variably expressed genes by median absolute deviation) compressed down into two length 100 vectors (mean and variance encoded spaces) which are made deterministic through the reparameterization trick of sampling an epsilon vector from the uniform distribution. The encoded layer is then decoded back to original 5000 dimensions through a single reconstruction layer. I included a layer of batch normalization in the encoding step to prevent dead nodes. The encoding scheme also uses relu activation while the decoder uses a sigmoid activation to enforce positive activations. All weights are glorot uniform initialized. 
+# 
+# Another trick used here to encourage manifold learning is _warm start_ as discussed in [Sonderby et al. 2016](https://arxiv.org/abs/1602.02282). With warm starts, we add a parameter _beta_, which controls the contribution of the KL divergence loss in the total VAE loss (reconstruction + (beta * KL)). In this setting, the model begins training deterministically as a vanilla autoencoder (_beta_ = 0) and slowly ramps up after each epoch linearly until _beta_ = 1. After a parameter sweep, we observed that kappa has little influence in training, therefore, we set  _kappa_ = 1, which is a full VAE.
+# 
+# Much of this script is inspired by the [keras variational_autoencoder.py example](https://github.com/fchollet/keras/blob/master/examples/variational_autoencoder.py)
+
+# In[1]:
+
+import os
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+
+import tensorflow as tf
+from keras.layers import Input, Dense, Lambda, Layer, Activation
+from keras.layers.normalization import BatchNormalization
+from keras.models import Model
+from keras import backend as K
+from keras import metrics, optimizers
+from keras.callbacks import Callback
+import keras
+
+import pydot
+import graphviz
+from keras.utils import plot_model
+from IPython.display import SVG
+from keras.utils.vis_utils import model_to_dot
+
+
+# In[2]:
+
+print(keras.__version__)
+tf.__version__
+
+
+# ## Load Functions and Classes
+# 
+# This will facilitate connections between layers and also custom hyperparameters
+
+# In[3]:
+
+# Function for reparameterization trick to make model differentiable
+def sampling(args):
+    
+    import tensorflow as tf
```

Comment:
Why the double import? Also if I am correct in this you mainly use keras. Do you need to have tensorflow imported globally or can you deal with this local import?